native/core/src/execution/datafusion/planner.rs

```diff
@@ -917,15 +917,11 @@ impl PhysicalPlanner {
 
                 let fetch = sort.fetch.map(|num| num as usize);
 
-                let copy_exec = if can_reuse_input_batch(&child) {
-                    Arc::new(CopyExec::new(child, CopyMode::UnpackOrDeepCopy))
-                } else {
-                    Arc::new(CopyExec::new(child, CopyMode::UnpackOrClone))
-                };
+                let child = Self::wrap_in_copy_exec(child);
```

Comment:
Can you add a similar comment here as you added with HashJoin on why sort needs a CopyExec?